qt-bugs@ issue : 166027
Trolltech task ID : none
bugs.kde.org number : none
applied: no
author: Lubos Lunak <l.lunak@kde.org>

This patch is just a hackish workaround that makes setting of WM_TRANSIENT_FOR
work with some window types (needed basically by the #0178 qt-copy patch).
It is not a proper solution, waiting for TT to come up with something.

--- a/src/gui/kernel/qwidget_x11.cpp
+++ b/src/gui/kernel/qwidget_x11.cpp
@@ -1648,7 +1648,7 @@
         do_size_hints(q, extra);
 
         // udpate WM_TRANSIENT_FOR
-        if (isTransient(q)) {
+        if (isTransient(q) && !( q->windowFlags() & Qt::X11BypassWindowManagerHint)) {
             QWidget *p = q->parentWidget();
             if (p)
                 p = p->window();
